semgrep/tests/e2e/rules/metavariable-regex-multi-regex.yaml

```diff
@@ -0,0 +1,13 @@
+rules:
+  - id: metavar-test-multi-regex
+    patterns:
+      - pattern: 'metavariable_regex_test.$X($Y)'
```

Comment:
I'd be curious to see a negative test case for:
```yaml
- pattern:
    metavariable: '$X'
    regex: 'method1'
```

Reply:
That gives me:

```
running 1 rules...
an internal error occured while invoking semgrep-core:
	invalid yaml: wrong rule fields
An error occurred while invoking the semgrep engine; please help us fix this by creating an issue at https://semgrep.dev
```

Is there something specific you're trying to test for here? This PR isn't as precise as it could be. Ideally we'd want to avoid all operators allowing `str|dict` and instead only allow specific operators to be their specific type, but that's a much larger effort and refactor.